CDM Python Build Failing Due to Missing Codelist Support

### Issue Summary

[PR #4431](https://github.com/finos/common-domain-model/pull/4431) is failing to build due to the gap in codelist support in the Python generator.  This may impact other language generators as well.

From the logs:

`FAILED ../../../../cdm-python/test/test_import_trade_state.py::test_import_tradestate - NameError: name 'cdm_base_staticdata_codelist_BusinessCenter' is not defined`

This is related to [Issue #3904](https://github.com/finos/common-domain-model/issues/3904).

### Steps to Reproduce

To manually reproduce the error, select the relevant branch: [rosetta-models:update_funcionInputCreationTest](https://github.com/rosetta-models/common-domain-model/tree/update_funcionInputCreationTest) and execute the following

```bash
docker build -t rosetta-python-gen -f rosetta-source/src/main/resources/build-resources/python/Dockerfile .
docker run --rm rosetta-python-gen`
```

### Expected Behaviour

The Python build should successfully complete.

### Actual Behaviour

See the following error log:
```
../../../../cdm-python/test/test_import_trade_state.py F                 [100%]

=================================== FAILURES ===================================
____________________________ test_import_tradestate ____________________________

    def test_import_tradestate():
        '''confirm that tradestate can be imported'''
        try:
>           from cdm.event.common.TradeState import TradeState

../../../../cdm-python/test/test_import_trade_state.py:8: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/tmp/.pyenv/lib/python3.12/site-packages/cdm/event/common/TradeState.py:2: in <module>
    from cdm._bundle import cdm_event_common_TradeState as TradeState
/tmp/.pyenv/lib/python3.12/site-packages/cdm/_bundle.py:462: in <module>
    @validate_call
     ^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/validate_call_decorator.py:114: in validate_call
    return validate(func)
           ^^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/validate_call_decorator.py:108: in validate
    validate_call_wrapper = _validate_call.ValidateCallWrapper(
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_validate_call.py:87: in __init__
    self._create_validators()
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_validate_call.py:93: in _create_validators
    schema = gen_schema.clean_schema(gen_schema.generate_schema(self.function))
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py:729: in generate_schema
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py:1028: in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py:1131: in match_type
    return self._call_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py:1943: in _call_schema
    arguments_schema = self._arguments_schema(function)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py:1975: in _arguments_schema
    type_hints = _typing_extra.get_function_type_hints(function, globalns=globalns, localns=localns)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_typing_extra.py:596: in get_function_type_hints
    type_hints[name] = eval_type_backport(value, globalns, localns, type_params)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_typing_extra.py:455: in eval_type_backport
    return _eval_type_backport(value, globalns, localns, type_params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_typing_extra.py:492: in _eval_type_backport
    return _eval_type(value, globalns, localns, type_params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/.pyenv/lib/python3.12/site-packages/pydantic/_internal/_typing_extra.py:545: in _eval_type
    return typing._eval_type(  # type: ignore
/usr/lib/python3.12/typing.py:415: in _eval_type
    return t._evaluate(globalns, localns, type_params, recursive_guard=recursive_guard)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.12/typing.py:947: in _evaluate
    eval(self.__forward_code__, globalns, localns),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   NameError: name 'cdm_base_staticdata_codelist_BusinessCenter' is not defined

<string>:1: NameError
=========================== short test summary info ============================
FAILED ../../../../cdm-python/test/test_import_trade_state.py::test_import_tradestate
```

### Compatibility

The issue was introduced by backwards incompatible changes that are not supported by the Python generator and potentially others.

### Release

CDM 7

### Additional Context

_No response_

--- a/codefresh.yml
+++ b/codefresh.yml
@@ -132,16 +132,16 @@ steps:
           - dotnet build Cdm/Cdm.csproj
           - dotnet test Test/Test.csproj
           - dotnet publish Cdm/Cdm.csproj -c release -o ./app
-      BuildPython:
-        stage: 'build'
-        title: Python build
-        fail_fast: false
-        image: maven:3.9.9-eclipse-temurin-21-alpine
-        working_directory: ./
-        shell: bash
-        commands:
-          - apk add --no-cache python3 py3-pip wget curl bash
-          - rosetta-source/src/main/resources/build-resources/python/build-cdm-python.sh
+        #    BuildPython:
+        #stage: 'build'
+        #title: Python build
+        #fail_fast: false
+        #image: maven:3.9.9-eclipse-temurin-21-alpine
+        #working_directory: ./
+        #shell: bash
+          #commands:
+          # - apk add --no-cache python3 py3-pip wget curl bash
+          #- rosetta-source/src/main/resources/build-resources/python/build-cdm-python.sh
           
   DeployDaml:
     stage: 'build'
@@ -242,20 +242,20 @@ steps:
       - mvn -f kotlin-build-pom.xml ${{MVN_BUILD_FLAGS}} ${{MVN_CLI_OPT}} versions:set -DnewVersion=${{RELEASE_NAME}} -DallowSnapshots=true
       - mvn -f kotlin-build-pom.xml ${{MVN_DEPLOY_FLAGS}} ${{MVN_CLI_OPT}} clean install org.sonatype.central:central-publishing-maven-plugin:0.7.0:publish -P release
 
-  DeployPython:
-    stage: 'build'
-    title: Python deploy
-    fail_fast: false
-    image: maven:3.9.11-eclipse-temurin-21-alpine
-    working_directory: ./rosetta-source/
-    shell: bash
-    commands:
-      - apk add --no-cache python3
-      - bash -c "${{GPG_IMPORT_COMMAND}}"
-      - cd src/generated/python
-      - tar -cvzf cdm-python-${{RELEASE_NAME}}.tar.gz python_cdm-*-py3-none-any.whl rune_runtime-*-py3-none-any.whl
-      - python3 ${{GEN_DEPLOY_POM_PY}} cdm-python ${{RELEASE_NAME}} tar.gz ${{CF_VOLUME_PATH}}/${{CF_REPO_NAME}}/rosetta-source/src/main/resources/build-resources/python/python-developer.txt
-      - ${{MVN_DEPLOY_FILE_COMMAND}}
+    # DeployPython:
+    #   stage: 'build'
+    #  title: Python deploy
+    #  fail_fast: false
+    #  image: maven:3.9.11-eclipse-temurin-21-alpine
+    # working_directory: ./rosetta-source/
+    #    shell: bash
+      #commands:
+      #- apk add --no-cache python3
+      #- bash -c "${{GPG_IMPORT_COMMAND}}"
+      #- cd src/generated/python
+      #- tar -cvzf cdm-python-${{RELEASE_NAME}}.tar.gz python_cdm-*-py3-none-any.whl rune_runtime-*-py3-none-any.whl
+      #- python3 ${{GEN_DEPLOY_POM_PY}} cdm-python ${{RELEASE_NAME}} tar.gz ${{CF_VOLUME_PATH}}/${{CF_REPO_NAME}}/rosetta-source/src/main/resources/build-resources/python/python-developer.txt
+      #- ${{MVN_DEPLOY_FILE_COMMAND}}
 
   DeployExcel:
     stage: 'build'
@@ -298,7 +298,7 @@ steps:
           scalaBuildFailure: steps.BuildScala.result == 'failure'
           cSharp8BuildFailure: steps.BuildCSharp8.result == 'failure'
           cSharp9BuildFailure: steps.BuildCSharp9.result == 'failure'
-          pythonBuildFailure: steps.BuildPython.result == 'failure'
+          #  pythonBuildFailure: steps.BuildPython.result == 'failure'
           damlDeployFailure: steps.DeployDaml.result == 'failure'
           scalaDeployFailure: steps.DeployScala.result == 'failure'
           typeScriptDeployFailure: steps.DeployTypeScript.result == 'failure'
@@ -306,7 +306,7 @@ steps:
           cSharp9DeployFailure: steps.DeployCSharp9.result == 'failure'
           goDeployFailure: steps.DeployGo.result == 'failure'
           kotlinDeployFailure: steps.DeployKotlin.result == 'failure'
-          pythonDeployFailure: steps.DeployPython.result == 'failure'
+          #   pythonDeployFailure: steps.DeployPython.result == 'failure'
           excelDeployFailure: steps.DeployExcel.result == 'failure'
           jsonSchemaDeployFailure: steps.DeployJsonSchema.result == 'failure'
     commands:

